File: examples/python/output/panel_alert.py
#!/usr/bin/env python3
"""Bordered panels and the alert presets.

    make run-example EX=python/output/panel_alert
"""

from __future__ import annotations

import sparcli as sc


def main() -> None:
    # Minimal panel.
    sc.panel('Hello from a panel.',
             sc.PanelOpts(border=sc.BorderStyle(sc.BorderType.SINGLE)))
    sc.println('')

    # Border, styled title (a Title object), padding, fixed width.
    sc.panel('Borders: single, double, rounded, thick.\n'
             'Multi-line content wraps into the frame.',
             sc.PanelOpts(
                 border=sc.BorderStyle(sc.BorderType.DOUBLE, sc.Color.CYAN),
                 title=sc.Title(text='Settings',
                                style=sc.Style.bold(sc.Color.CYAN),
                                halign=sc.Align.LEFT, pad=1),
                 padding=sc.Edges(1, 2, 1, 2),
                 width=56))
    sc.println('')

    # Full width, centered content.
    sc.panel('This panel spans the whole terminal.',
             sc.PanelOpts(border=sc.BorderStyle(sc.BorderType.ROUNDED),
                          full_width=True, content_align=sc.Align.CENTER))
    sc.println('')

    # Alert presets: panel wrappers with a preset icon + color.
    sc.alert.info('Service started on port 8080.')
    sc.alert.debug('Cache warmed in 132 ms.')
    sc.alert.warning('Disk usage at 85 %.')
    sc.alert.error('Connection to database lost.\nRetrying in 5 s.')
    sc.alert.success('All 128 tests passed.')


if __name__ == '__main__':
    main()

File: examples/rust/output/panel_alert.rs
//! Bordered panels and the alert presets.
//!
//!   cargo run -p sparcli --example output_panel_alert

use sparcli::*;

fn main() {
    // Minimal panel.
    panel("Hello from a panel.", PanelOpts::new().single());
    println("", Style::default());

    // Border, title, full width, centered content.
    panel(
        "Borders: single, double, rounded, thick.\n\
         Multi-line content wraps into the frame.",
        PanelOpts::new()
            .border(BorderStyle::new(BorderType::Double).color(Color::CYAN))
            .title("Settings")
            .content_align(Align::Center)
            .full_width(),
    );
    println("", Style::default());

    // Colored border + background.
    panel(
        "Colored border with a dark background.",
        PanelOpts::new()
            .border(BorderStyle::new(BorderType::Rounded).color(Color::BLUE))
            .bg(Color::rgb(20, 20, 40))
            .title("styled"),
    );
    println("", Style::default());

    // Alert presets: panel wrappers with a preset icon + color.
    alert::info("Service started on port 8080.");
    alert::debug("Cache warmed in 132 ms.");
    alert::warning("Disk usage at 85 %.");
    alert::error("Connection to database lost.\nRetrying in 5 s.");
    alert::success("All 128 tests passed.");
}
